deepsearch/documents/core/input_process.py

```diff
@@ -1,173 +1,118 @@
-import datetime
-import glob
 import logging
 import os
-import pathlib
-import zipfile as z
+import tempfile
 from pathlib import Path
 from typing import Any, List
 
 import urllib3
 from tqdm import tqdm
 
 from deepsearch.cps.client.api import CpsApi
+from deepsearch.cps.client.components.documents import DocumentConversionResult
+from .utils import batch_single_files, create_root_dir
 
 urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
 
-from .common_routines import ERROR_MSG, progressbar_padding, success_message
+from .common_routines import success_message
 from .convert import (
     check_status_running_tasks,
-    download_converted_docs,
+    download_converted_documents,
+    get_download_url,
     send_files_for_conversion,
     send_urls_for_conversion,
 )
 from .create_report import report_docs, report_urls
 
 logger = logging.getLogger(__name__)
 
-# set up basic urls
-url_user_management = "/user/v1"
-url_public_apis = "/public/v4"
 
-
-def process_local_input(api: CpsApi, cps_proj_key: str, local_file: Path):
+def process_local_input(
+    api: CpsApi, cps_proj_key: str, source_path: Path, progress_bar=False, cli_use=False
+) -> DocumentConversionResult:
     """
     Classify the user provided local input and take appropriate action.
     """
-    if not os.path.exists(local_file):
+    if not os.path.exists(source_path):
         logger.error("Error: File not found. Check input.")
     else:
-        root_dir = create_root_dir()
-        batched_files = batch_single_files(local_file=local_file, root_dir=root_dir)
-        task_ids = send_files_for_conversion(
-            api=api, cps_proj_key=cps_proj_key, local_file=local_file, root_dir=root_dir
+        with tempfile.TemporaryDirectory() as tmpdir:
+            batched_files = batch_single_files(
+                source_path=source_path,
+                root_dir=Path(tmpdir),
+                progress_bar=progress_bar,
+            )
+            task_ids = send_files_for_conversion(
+                api=api,
+                cps_proj_key=cps_proj_key,
+                source_path=source_path,
+                root_dir=Path(tmpdir),
+                progress_bar=progress_bar,
+            )
+            statuses = check_status_running_tasks(
+                api=api,
+                cps_proj_key=cps_proj_key,
+                task_ids=task_ids,
+                progress_bar=progress_bar,
+            )
+        if cli_use == False:
```

Comment:
Sorry, we can't do this. This is in a library function and it must not need _any_ knowledge that a CLI even exists.
The CLI implementation should simply use the library classes/method